docs/data/upload.md

```diff
@@ -0,0 +1,113 @@
+   The following example code could be placed inside the `main()` function (or a function called from `main()`):
+
+   ```python {class="line-numbers linkable-line-numbers"}
+   await data_client.file_upload_from_path(
+     # The ID of the machine part the file should be associated with
+     part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+     # Any tags you want to apply to this file
+     tags=["cat", "animals", "brown"],
+     # Path to the file
+     filepath="/Users/Artoo/my_cat_photos/brown-cat-on-a-couch.png"
+   )
+   ```
+
+   To upload all the files in a directory, you can use the same [`file_upload_from_path`](/build/program/apis/data-client/#fileuploadfrompath) method inside a `for` loop, for example:
+
+   ```python {class="line-numbers linkable-line-numbers"}
+   import os # Add this package up at the top with your other imports
+
+   my_data_directory = "/Users/Artoo/my_cat_photos"
+
+   for file_name in os.listdir(my_data_directory):
+     await data_client.file_upload_from_path(
+       part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+       tags=["cat", "animals", "brown"],
+       filepath=os.path.join(my_data_directory, file_name)
+     )
+   ```
+
+4. Save and run your code once.
+   Running your code more than once will duplicate the data.
+   View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+## Sync data with the `sync` method
+
+The Viam Golang SDK includes a data service API with a [`sync`](/data/#sync) method that syncs all the data from your machine to the cloud.
+The [cloud sync tools](/data/cloud-sync/) sync data automatically according to how you configure cloud sync, but if you prefer to sync programatically, you can instead run a script that calls the `sync` method.
+
+The following steps assume you already have a [data capture service configured](/data/capture/#add-the-data-management-service).
+If you want to use the `sync` API method as the only way to sync data, you should toggle sync off in your data service config (`"sync_disabled": true` in raw JSON).
+
+1. Install the [Viam Go SDK](https://github.com/viamrobotics/rdk/tree/main/robot/client).
+
+2. Create a Go script file in a directory of your choice.
+   Go to your machine's **Code sample** tab in the [Viam app](https://app.viam.com) and click **Golang**.
+   Copy the code from that tab into your script file to establish a connection from your computer to your machine's instance in the Viam app.
+
+3. Add the required import to the list of imports at the top of your script:
+
+   ```go {class="line-numbers linkable-line-numbers"}
+   import (
+     "go.viam.com/rdk/services/datamanager"
+   )
+   ```
+
+4. In your `main()` function (or in a function called from `main`), call [the `sync` method as documented](/data/#sync).
```

Comment:
Nit: full task description in link title, 

```suggestion
4. In your `main()` function (or in a function called from `main()`), [call the `sync` method](/data/#sync).
```

Or:

```suggestion
4. In your `main()` function (or in a function called from `main()`), call the [`sync` method](/data/#sync).
```

I feel that `as documented` is readily-inferable from the context, and is explicitly demonstrated in the sample below.